index.d.ts

```diff
@@ -0,0 +1,16 @@
+export type Scalar = number | string | boolean | bigint | null | undefined;
+export interface Err<Code extends Scalar, Ext extends Object> extends Error, Ext {
+  code: Code
+}
+
+interface CreateError {
+  <Code extends Scalar, Ext extends Object> (
+    reason:string|Error,
```

Comment:
This should just be `Error`, the string detection is done in the code to encourage users to move off the v1 API but maybe now it's time to demand an `Error`.